static_frame/core/frame.py

```diff
@@ -1493,17 +1495,17 @@ def from_sql(cls,
             {name}
             {consolidate_blocks}
         '''
-        row_gen = connection.execute(query)
+        sql_iterable = connection.execute(query)
 
         own_columns = False
         columns = None
         if columns_depth == 1:
-            columns = cls._COLUMNS_CONSTRUCTOR(b[0] for b in row_gen.description[index_depth:])
+            columns = cls._COLUMNS_CONSTRUCTOR(b[0] for b in sql_iterable.description[index_depth:])
             own_columns = True
         elif columns_depth > 1:
             # use IH: get via static attr of columns const
             constructor = cls._COLUMNS_HIERARCHY_CONSTRUCTOR.from_labels_delimited
-            labels = (b[0] for b in row_gen.description[index_depth:])
+            labels = (b[0] for b in sql_iterable.description[index_depth:])
```

Comment:
I wonder if there is a way we can use `columns_select` here, in the creation of columns, rather than having to trim columns later after creating columns here. That way we save creating potentially huge `Index` if the user only is requesting a single column. We might use `sql_iterable.description` and `columns_select` to also determine the appropriate indexes to extract with `itemgetter`, below, saving us from having to use the columns to determine that.

Reply:
Good point. I'll optimize this.